src/components/InventoryManager/ListingInputField.js

```diff
@@ -1,92 +1,264 @@
-        },
-      };
-    } else if (type === 'number') {
-      InputProps = {
-        inputProps: {
-          min: 0,
         },
       };
     }
     return InputProps;
   }
   const inputProps = getInputProps({});
-  if (multiline) {
+  if (type === 'currency') {
+    return (
+      <>
+        <CurrencyTextField
+          variant="outlined"
+          placeholder="0.00"
+          currencySymbol="$"
+          value={val}
+          decimalCharacter="."
+          onChange={onChange}
+          textAlign="left"
+          size="small"
+        />
+      </>
+    );
+  }
+  if (type === 'autoComplete') {
     return (
       <>
-        <Typography variant="h6" component="h6">{label}</Typography>
+        <Grid item xs={12}>
+          <Typography className={classes.infoLabel} variant="h2" component="h2">{label}</Typography>
+        </Grid>
+        <Grid item xs={12}>
+          <Autocomplete
+            autoComplete
+            options={options}
+            getOptionLabel={getLabel}
+            onChange={onChange}
+            value={val}
+            renderInput={(params) => (
+              <TextField
+                name={name}
+                {...params}
+                placeholder={placeholder}
+                variant="outlined"
+                required
+              />
+            )}
+            classes={{ input: classes.text }}
+          />
+        </Grid>
+      </>
+    );
+  }
+
+  if (type === 'multiline') {
+    return (
+      <>
+        <Grid item xs={12}>
+          <Typography className={classes.infoLabel} variant="h2" component="h2">{label}</Typography>
+        </Grid>
+        <Grid item xs={12}>
+          <TextField
+            required
+            multiline
+            rows={3}
+            id={id}
+            name={name}
+            type={type}
+            value={val}
+            InputProps={inputProps}
+            onChange={onChange}
+            className={classes.inputField}
+            variant="outlined"
+            placeholder={placeholder}
+            fullWidth
+          />
+        </Grid>
+      </>
+    );
+  }
+
+  if (type === 'number') {
+    return (
+      <>
+        <IconButton className={classes.button} onClick={() => onButtonClick(name, -1)}>
+          <RemoveIcon fontSize="small" />
+        </IconButton>
         <TextField
-          required
-          multiline
-          rows={3}
           id={id}
           name={name}
-          type={type}
           value={val}
           InputProps={inputProps}
           onChange={onChange}
-          className={classes.inputField}
+          className={classes.numberInputField}
           variant="outlined"
+          size="small"
         />
+        <IconButton className={classes.button} onClick={() => onButtonClick(name, 1)}>
+          <AddIcon fontSize="small" />
+        </IconButton>
+      </>
+    );
+  }
+
+  if (type === 'date') {
+    const focusTextField = () => {
+      textFieldRef.current.click();
```

Comment:
oooh it looks like this is because Google Chrome has the calendar icon decoration in the textfield, but Firefox does not.
I'll take out the icon I added then